utils/snapshot_extractor.mjs

```diff
@@ -0,0 +1,83 @@
+// @format
+import "dotenv/config";
+import { resolve, dirname } from "path";
+import { fileURLToPath } from "url";
+import { once } from "events";
+import { Worker } from "worker_threads";
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+export default async function snapshotExtractor2(
+  extractor,
+  { inputs, expect }
+) {
+  const map = new Map();
+  expect.write.forEach((line) => map.set(line, false));
+
+  const worker = new Worker(resolve(__dirname, "./worker_start.mjs"), {
+    workerData: {
+      concurrency: 20,
+    },
+  });
+
+  let numberOfPostedMessages = 0;
+  const postMessage = (message) => {
+    numberOfPostedMessages++;
+    worker.postMessage(message);
+  };
+
+  // Uncomment this to see what is being written
+  const DEBUG = false;
+  let writeResult = "";
+  const write = (string) => {
+    if (DEBUG && string) {
+      writeResult += string + "\n";
+    }
+  };
+
+  worker.on("message", (message) => {
+    if (message.error) {
+      worker.emit("exit");
+      throw new Error(message.error);
+    }
+
+    const ret = extractor.update(message);
```

Comment:
Is `ret` short for `return`?

Reply:
Yes

Reply:
I think "return" is a too generic name. Any function's output is a return so using this term isn't scalable IMO. Better would be to identify what is being returned.

Reply:
I think the ret works because I am not using ret directly. I am writing ret.messages or ret.write. At first, I was destructuring it to `const {messages, write} = extractor.update` but the `write` and `messages` keyword are too common. It was creating a confusion. I also have a function named write.

> Better would be to identify what is being returned.

There is no single word for what `update` returns. We can call it data.

Reply:
no it's fine let's call it ret